mesonbuild/compilers.py

```diff
@@ -1639,7 +1639,8 @@ def get_std_shared_lib_link_args(self):
         return ['-shared']
 
     def get_soname_args(self, prefix, shlib_name, suffix, path, soversion, is_shared_module):
-        return []
+        # FIXME: Make this work for Windows, MacOS and cross-compiling
+        return get_gcc_soname_args(GCC_STANDARD, prefix, shlib_name, suffix, path, soversion, is_shared_module)
```

Comment:
This line is copied almost verbatim, so we would need to adjust a lot of other places to fit that requirement as well ;-)